### PR TITLE
[MISSING TESTS] Adds manageSubscriptionURLForProduct

### PR DESCRIPTION
**Pull request by vegaro**
Adds a function `- (void)manageSubscriptionURLForProduct:(SKProduct *)product withCompletionBlock:(void (^)(NSURL *))completion;` that returns a NSURL that can be opened to manage the account subscriptions.

### CODE REVIEW

**Comment by vegaro** [comment]
@jeiting Missing tests, but what do you think of the API?

File: Purchases/Public/RCPurchases.m [hunk 1]
```diff
@@ -632,6 +632,18 @@ - (void)paymentDiscountForProductDiscount:(SKProductDiscount *)discount
     }];
 }
 
+- (void)manageSubscriptionURLForProduct:(SKProduct *)product
+                    withCompletionBlock:(void (^)(NSURL *))completion
+{
+    [self.backend manageSubscriptionsURLForProductID:product.productIdentifier withAppUserID:self.appUserID completion:^(NSURL * _Nullable url, NSError * _Nullable error) {
+        if (error) {
+            RCDebugLog(@"There was an error fetching URL, using default.");
+            url = [NSURL URLWithString:@"itms-apps://apps.apple.com/account/subscriptions"];
```
Comment (aboedo) [inline Purchases/Public/RCPurchases.m]: nitpick: extract magic string

File: Purchases/RCBackend.m [hunk 2]
```diff
@@ -459,4 +458,30 @@ - (void)postOfferForSigning:(NSString *)offerIdentifier
                   }];
 }
 
+- (void)manageSubscriptionsURLForProductID:(NSString *)productID
+                             withAppUserID:(NSString *)appUserID
+                                completion:(nullable void (^)(NSURL * _Nullable url, NSError * _Nullable error))completion
+{
+    NSString *escapedAppUserID = [self escapedAppUserID:appUserID];
+    NSString *path = [NSString stringWithFormat:@"/subscribers/%@/subscriptions/%@/manage", escapedAppUserID, productID];
```
Comment (aboedo) [inline Purchases/RCBackend.m]: nitpick: extract magic string

File: Purchases/RCBackend.m [hunk 3]
```diff
@@ -459,4 +458,30 @@ - (void)postOfferForSigning:(NSString *)offerIdentifier
...
+    [self.httpClient performRequest:@"GET"
+                               path:path
+                               body:nil
+                            headers:self.headers
+                  completionHandler:^(NSInteger statusCode, NSDictionary *_Nullable response, NSError *_Nullable error) {
+                        if (error == nil && statusCode < 300) {
```
Comment (aboedo) [inline Purchases/RCBackend.m]: nitpick: extract magic numbers

**Comment by aboedo** [comment]
other than rebasing, are there any holdups for this one?

**Comment by vegaro** [comment]
Yeah, I am going to close it because the implementation in the backend totally changed. This needs to be rebuilt.

**Pull request closed without merging**